### PR TITLE
Add test for CoreImpl::get_versions()

### PR DESCRIPTION
**Pull request by awayzjj**
Closes [23298](https://github.com/openvinotoolkit/openvino/issues/23298)
### Tickets: 
- [CVS-132140](https://jira.devtools.intel.com/browse/CVS-132140)

### CODE REVIEW

**Review by vurusovs — APPROVED** [review]

File: src/inference/tests/functional/ov_register_plugin_test.cpp [hunk 1]
```diff
@@ -53,6 +53,8 @@ TEST(RegisterPluginTests, getVersionforRegisteredPluginThrows) {
 TEST(RegisterPluginTests, getVersionforNoRegisteredPluginNoThrows) {
     ov::Core core;
     ASSERT_NO_THROW(core.get_versions("unkown_device"));
+    std::map<std::string, ov::Version> versions = core.get_versions("unkown_device");
+    ASSERT_TRUE(versions.empty());
```
Comment (vurusovs) [inline src/inference/tests/functional/ov_register_plugin_test.cpp:57]:
```suggestion
    std::map<std::string, ov::Version> versions;
    ASSERT_NO_THROW(versions = core.get_versions("unkown_device"));
    ASSERT_TRUE(versions.empty());
```

Could you optimize number of `get_versions()` calls?

Comment (awayzjj) [reply src/inference/tests/functional/ov_register_plugin_test.cpp]: Thank you for your review. I fixed it as per your suggestion.

**Comment by vurusovs** [comment]
build_jenkins

**Review by olpipi — APPROVED** [review]

**Comment by olpipi** [comment]
build_jenkins

**Comment by awayzjj** [comment]
@vurusovs @olpipi  Hi, I don not know why ci/jenkins shows the commit has test failures,  the Details link is not available, could you give me some advice on finding out the failed test case?

**Comment by olpipi** [comment]
> @vurusovs @olpipi Hi, I don not know why ci/jenkins shows the commit has test failures, the Details link is not available, could you give me some advice on finding out the failed test case?

This is a false positive case. All checks passed but overall status is failed. 
@ilya-lavrenov could you please merge this PR?

**Pull request merged**